### ISSUE TITLE
PLR1716 fix introduces a syntax error with unbalanced parentheses

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#13524]
When the two expressions to be collapsed by [`boolean-chained-comparison` (PLR1716)](https://docs.astral.sh/ruff/rules/boolean-chained-comparison/) are parenthesized with different numbers of parentheses, the fix makes Ruff fail.

```console
$ ruff --version
ruff 0.6.8
$ echo '(a < b) and b < c' | ruff check --isolated --preview --select PLR1716 - --fix

error: Fix introduced a syntax error. Reverting all changes.

This indicates a bug in Ruff. If you could open an issue at:

    https://github.com/astral-sh/ruff/issues/new?title=%5BFix%20error%5D

...quoting the contents of `-`, the rule codes PLR1716, along with the `pyproject.toml` settings and executed command, we'd be very appreciative!

(a < b) and b < c
-:1:2: PLR1716 Contains chained boolean comparison that can be simplified
  |
1 | (a < b) and b < c
  |  ^^^^^^^^^^^^^^^^ PLR1716
  |
  = help: Use a single compare expression

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

**Comment by zanieb** [comment]
Ah this looks like.. hard to fix. We can at least not offer an invalid fix here.

**Comment by dscorbett** [comment]
This was addressed in version 0.6.9. Is there anything left to do for this issue?

**Comment by zanieb** [comment]
We don't offer the fix, but we can do better and offer the fix with proper handling.
